### DISCUSSION TITLE
Help needed: Can't add Webparts on SharePoint page

### DISCUSSION THREAD

**Discussion by steffen224** [microsoft-search/pnp-modern-search#2719] [General]
Hello,

we have some issues on one of our tenants. We deployed the app package over the SharePoint App Catalog and installed the App on the SiteCollection. But we are not able to add the webparts on a page, because the webparts aren't show up. We also tested a deployment on a SiteCollection AppCatalog. We don't get any error messages or something like that. The issue occurs on every new SiteCollection we create.
Do you have any idea?

Best regards and thank you!

**Comment by kasperbolarsen** [comment]
Was it 4.8.0 you installed?  And did it ask for additional graph permissions?

**Reply by steffen224** [reply]
We first installed 4.7.0 but then updated to 4.8.0, because we thought that this maybe fix our problem. We didn't grant any graph permissions, so I have to check if it will ask for these additional graph permissions. I will get back to you today.

Thank you very much!

**Reply by wobba** [reply]
Graph permissions should not matter. If you did an upgrade you should remove 4.7 and re-add with 4.8 due to a revert in packaging mentioned in the release notes.

**Comment by fisketryne5** [comment]
@steffen224 Have you added the app to the site? 
![image](https://user-images.githubusercontent.com/22908383/216364640-40f8d0c4-c3ec-4b3f-a603-8bc2b3f57977.png)

**Comment by CasCabrera** [comment]
I'm running into a similar issue where I've added/uploaded the PNP Modern Search V4.8.0 app into my sharepoint tenant and enabled it across the org as well as allowed/permitted all the permission requests that came along with the install. 

I followed this guide as a reference:

https://microsoft-search.github.io/pnp-modern-search/installation/

When I go to add the web part into a sharepoint site I get the following error message:

![Screenshot 2023-04-28 at 2 28 28 PM](https://user-images.githubusercontent.com/132079549/235225553-7bcfedd7-ab70-41d9-bfbc-ace02a768c47.png)
![Screenshot 2023-04-28 at 2 17 26 PM](https://user-images.githubusercontent.com/132079549/235225554-9b580ce0-641b-4fe4-813b-8e19bc904137.png)

I'm not sure if there are other dependencies that need to be installed, but I followed the above guide where it does not mention any dependencies that need to be installed.

Thanks for looking into this!

**Reply by wobba** [reply]
Did you read the release notes? https://github.com/microsoft-search/pnp-modern-search/releases/tag/4.8.0 Did you upgrade from a previous version?

**Reply by CasCabrera** [reply]
Hello @wobba, thank you for the response. I am new to all this and did indeed go through the documentation. As far as I can tell, all I would need to do is add the PNP Modern Search app to my SharePoint online tenant and enable it for all my sites or individual sites, add the webpart on my search results site, do some tweaking and it should theoretically work. I must be missing a step somewhere because this theoretically should work.

I am not upgrading, this is a fresh new install of the PNP Modern Search app.

I was reading something about the SharePoint's Framework version needing to be v1.15.2 for existing extensions. But again this is a new install, I don't believe this pertains to my scenario, but I could be wrong. Any help on this would be greatly appreciated.

Thanks, Cas.

**Reply by wobba** [reply]
If you install in an app catalog and then add to a site it should "just work". I would try to delete from the app catalog and re-add it there and see if that helps. No need to re-add on pages.

**Reply by CasCabrera** [reply]
Good day Wobba, 

That is precisely where I am uploading the app to. I tried uploading the app via the app catalog route by manually uploading the app, by drag and dropping it into the app catalog.

The issue I'm seeing, no matter how many times I've removed and added the app via the many methods mentioned in the step by step tutorial, the app simply does not appear when trying to add it to a site. 

The app shows up in the SharePoint online catalog after uploading it, but when I go to the site I want to use it in the app does not show up. When I go to add the webpart on the site I see the four modern search webparts and can pick it, but run into the issue above. 

Is it possible the app is simply not supported with the instance of SharePoint online we have?

We have another app that was added without issues on our end.

![image](https://user-images.githubusercontent.com/132079549/236033211-a61ba4fa-5499-46cf-ba55-d8a2184d539f.png)

Thanks,

Cas

**Reply by wobba** [reply]
Nothing special with the web parts. And you installed the app to be available to all sites? Or do you need to add it per site first?

**Reply by CasCabrera** [reply]
Hello @wobba, I installed the app in the SharePoint app catalog, when prompted to enable for all sites, I clicked on enable. When all the permission requests came in, I approved them.

So as it appears the issue isn't uploading or installing the app, the issue is when we go to the SharePoint site to add an app to the site, it doesn't show up. Then when we go to add the web part in our search results site page, we run into the error I mentioned in my original post.

I assume, since the app isn't showing up on my apps list for the site I'm trying to configure, the webpart is not going to work. It's just odd that we enabled the app for all sites and approved all permissions requests, yet it does not show up and or work.

Just to rule out the site I'm working with as a possible factor, I created a new SharePoint site, created a new results page, tried to add the app to the site, it does not show up. Tried to add the Web Part to the page, but I still get the error message mentioned originally.

![image](https://user-images.githubusercontent.com/132079549/236297649-05bc3382-7316-4f7d-a4d8-c5cf3179db9a.png)

![image](https://user-images.githubusercontent.com/132079549/236297757-74fbf938-0406-476d-9172-b270194dfc55.png)

Thanks,

Cas.

**Reply by kasperbolarsen** [reply]
Hi @CasCabrera , are you on the common kind of tenant,  not GCC or GCC high?

**Reply by CasCabrera** [reply]
Hello @kasperbolarsen, 

I have a common tenant, certainly not GCC or GCC high. 

Thanks,

Cas

**Reply by wobba** [reply]
"So as it appears the issue isn't uploading or installing the app, the issue is when we go to the SharePoint site to add an app to the site, it doesn't show up. " <- when approved for all sites, there is no app to add. The web parts should just show up for all users on all pages.

And you get the same error if adding the webpart to a page now?

**Reply by CasCabrera** [reply]
@wobba that is correct. We are able to upload the app into our SharePoint instance, but unable to get the web part working. When we try to add web part into our results page, we get the error mentioned early in my original post. I have a ticket open with MS support to see if they are able to assist with this.

Once we get past the web part not working, I'll respond with the solution to my issue. 

Thanks again for all your help.

-Cas

**Reply by wobba** [reply]
Can you remove the app from the app catalog. Then navigate to `<app catalog site url>/Lists/ComponentManifests/AllItems.aspx`. If there are entries for the pnp search web parts, remove them. Then re-add the sppkg to the app catalog.

If you don't use the Graph API, then you don't have to approve the scopes.

**Reply by CasCabrera** [reply]
Hi @wobba,

I removed the app from the app catalog and navigated to the page referenced and did not see any entries for the pnp search web parts. 

I then proceeded to re-add the sspkg to the app catalog, but still run into the same issue when trying to add a web part to the site. 

I do see some references for the pnp modern search results after I add the app to my app catalog and then navigate to the above referenced page, it just does not work when trying to add as a web part to my site.

![image](https://user-images.githubusercontent.com/132079549/236561446-12864036-4772-489e-8e84-3e6b3337146e.png)

Thanks, 

Cas

**Reply by wobba** [reply]
Can you navigate to `sites/appcatalog/clientsideassets/59903278-dd5d-4e9e-bef6-562aae716b8b` and look for the CommonStrings file you get an error on, and check the name? I can see it in my install.
<img width="484" alt="image" src="https://user-images.githubusercontent.com/51104/236567936-71e5f0df-1f32-4400-80d9-db02bcd46991.png">

**Reply by CasCabrera** [reply]
Hello @wobba I do see this file mentioned. Anything in particular you're looking for in this file?

**Reply by wobba** [reply]
If the file is there, then you should not get the error you get saying it cannot find the file. Don't know how to help you troubleshoot. You can always trying to enable CDN which is always a good idea anyways. See https://learn.microsoft.com/en-us/sharepoint/dev/spfx/enable-microsoft-365-content-delivery-network?tabs=msposh#enable-public-cdn-in-the-tenant

![image](https://github.com/microsoft-search/pnp-modern-search/assets/51104/97a9cf6e-bb61-4ff0-90ca-3205803843b0)

**Reply by CasCabrera** [reply]
Hello @wobba looks like the issue was related to my admin account not having proper licensing to add the webparts to the sites pages. 

After getting on a call with MS support we were able to determine the issue came down to my admin account not being properly licensed. We tried with an account that had the correct licensing and we were able to add the webparts without issues. 

Thanks for taking the time to look into this. Really appreciate the time, effort and attention to detail you provided.

Thanks,

Cas.

**Reply by wobba** [reply]
Happy you got it resolved, and always take care when using admin accounts. Even though they have SP access, doesn't mean they have read access by default everywhere :)
